examples/minimal-appdir/src/app/page.tsx

```diff
@@ -70,6 +71,44 @@ export default function Home() {
           await startUpload(files);
         }}
       />
+      <UT.Root
+        endpoint="videoAndImage"
+        onClientUploadComplete={(res) => {
+          console.log(`onClientUploadComplete`, res);
+          alert("Upload Completed");
+        }}
+        onUploadBegin={() => {
+          console.log("upload begin");
+        }}
+      >
+        <UT.Dropzone style={{ marginTop: 24 }}>
+          {({ dropzone, isUploading }) => (
+            <div
+              style={{
+                borderWidth: 2,
+                borderStyle: "dashed",
+                borderColor: dropzone?.isDragActive ? "#2563f5" : "#11182725",
+                padding: 16,
+              }}
+            >
+              <p
+                style={{
+                  width: "fit-content",
+                }}
+              >
+                Drag and drop
+              </p>
+              <UT.Button as="button">
+                {isUploading ? "Uploading" : "Upload file"}
+              </UT.Button>
+              <UT.AllowedContent
+                as="p"
+                style={{ fontSize: 12, width: "fit-content" }}
+              />
```

Comment:
how would one do content customization here? should it optionally take a render function that gets the file types & multiple as props so they can choose to reformat the text?

Reply:
It already has an optional render function with all of this stuff. It currently exposes the entire context which includes `fileTypes`:
```ts
type PrimitiveContextValues = {
  ...
  fileTypes: FileRouterInputKey[];
  ...
};
```